notebooks/214-vision-PaddleCls/214-vision-PaddleCls.ipynb

```diff
@@ -0,0 +1,287 @@
+    "    urllib.request.urlretrieve(mobilenet_url, \"model/MobileNetV3_large_x1_0_infer.tar\")\n",
+    "    print(\"Model Downloaded\")\n",
+    "\n",
+    "    file = tarfile.open(\"model/MobileNetV3_large_x1_0_infer.tar\")\n",
+    "    res = file.extractall('model')\n",
+    "    file.close()\n",
+    "    if (not res):\n",
+    "        print(\"Model Extracted to \\\"model/MobileNetV3_large_x1_0_infer\\\".\")\n",
+    "    else:\n",
+    "        print(\"Error Extracting the model. Please check the network.\")"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "f3a72a48-494e-44f3-9ee8-b2c15c845fdc",
+   "metadata": {},
+   "source": [
+    "# Define the callback function for postprecessing"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "36b52bfa-551c-4d58-a796-7a0cb7e8c357",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "def callback(infer_request, i) -> None:\n",
+    "    imagenet_classes = json.loads(open(\"utils/imagenet_class_index.json\").read())\n",
+    "    predictions = next(iter(infer_request.results.values()))\n",
+    "    indices = np.argsort(-predictions[0])\n",
+    "    if (i == 0):\n",
+    "        # Calculate the first inference time\n",
+    "        latency = time.time() - start\n",
+    "        print(\"latency:\", + latency)\n",
+    "        for i in range(5):\n",
+    "            print(\n",
+    "                \"Class name:\",\"'\" + imagenet_classes[str(list(indices)[i])][1] + \"'\",\n",
+    "                \", probability:\" , predictions[0][list(indices)[i]])"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "c8adfba1-82d2-4526-910f-2d32530d74eb",
+   "metadata": {},
+   "source": [
+    "# Read the model file "
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "179ae1a7-966d-49fe-9c06-811257c57989",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "import openvino.runtime as ov\n",
+    "\n",
+    "core = ov.Core()\n",
+    "# MobileNetV3_large_x1_0\n",
+    "model = core.read_model(\"model/MobileNetV3_large_x1_0_infer/inference.pdmodel\")\n",
+    "# get the information of intput and output layer\n",
+    "input_layer = next(iter(model.inputs))\n",
+    "output_layer = next(iter(model.outputs))"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "fc6a1ee1-802b-4561-af31-a613bba10fc6",
+   "metadata": {},
+   "source": [
+    "# Integrate prepocessing steps into execution graph with Preprocessing API\n",
+    "When your input data don’t perfectly fit to Neural Network model input tensor - this means that additional operations/steps are needed to transform your data to format expected by model. These operations are known as “preprocessing”.\n",
+    "Preprocessing steps will be integrated into execution graph and will be performed on selected device (CPU/GPU/VPU/etc.) rather than always being executed on CPU. This will improve selected device utilization which is always good."
```

Comment:
```suggestion
    "Preprocessing steps are integrated into the execution graph and performed on the selected device(s) (CPU/GPU/VPU/etc.) rather than always executed on CPU. This improves utilization on the selected device(s)."
```